packages/lit-html/src/lib/lit-html.ts

```diff
@@ -233,12 +233,17 @@ class Template {
           // HTML, like `<?` or `</{`. Figure out whether / how to handle that.
           if (match[1] === '!--') {
             regex = commentRegex;
-          } else {
+          } else if (match[2] === undefined) {
+            // If match[2] !== undefined, then we have an unescaped < and stay
+            // in the text state
             if (rawTextElement.test(match[1])) {
-              regex = new RegExp(`<\/${match[1]}`);
+              regex = new RegExp(`<\/${match[1]}`, 'g');
             } else {
               regex = tagRegex;
             }
+          } else if (match[2] === '' || match[2] === '/') {
```

Comment:
I think we need to assert that the `/` here is directly before a interpolation? But the current regex doesn't guarantee that. It could just be a `abc </ def`, which I don't think should count as a dynamic tag.

Reply:
And once the regex is fixed above, can the final `else if` just be an `else`?

Reply:
I changed `textRegex` so that each type of match that ends the state is detected in its own group, so the `/` assertion is taken care of in the regex itself now. The case `abc </ def` actually ends up being parsed as `abc <!-- def-->` by HTML, and we match that behavior now.

Now I'm not sure how _good_ that is for us to do, since it should be very rare and fixable when that happens and we probably don't need to pay for the bytes to detect it, but at this point I'd like to move forward with a little more code to be more correct, and take a little longer and deliberate look at loosening up the parsing and documenting where we differ from spec-compliant HTML. I'm particularly curious whether we can do better parsing in the upcoming dev mode and issue warnings for no cost in prod mode.